yt/data_objects/time_series.py

```diff
@@ -274,14 +274,22 @@ def piter(self, storage = None):
         ...
 
         """
-        dynamic = False
         if self.parallel is False:
             njobs = 1
-        else:
+        elif dynamic is False:
             if self.parallel is True:
                 njobs = -1
             else:
                 njobs = self.parallel
+        else:
+            my_communicator = communication_system.communicators[-1]
+            nsize = my_communicator.size
+            if nsize == 1:
+                self.parallel = False
+                dynamic = False
+                njobs = 1
+            else:
+                njobs = nsize - 1
```

Comment:
@jwise77, is it necessary to set `njobs` this way here? This also happens inside `dynamic_parallel_objects`, which is the function called by `parallel_objects` when `dynamic=True`. This might be removing one too many jobs.

Reply:
I looked at this, and `dynamic_parallel_objects` only removes one process if `njobs` isn't set. But since we're passing in `njobs`, we have to subtract 1. I blindly tried to remove the "-1" and I got this error message,
`P001 yt : [ERROR    ] 2019-02-26 09:38:05,303 You have asked for 4 jobs, but only 3 processors are available.`
So I'm keeping this -1 in there.